### DISCUSSION TITLE
[RFC] Semantic DOM Structure with classNames or styles

### DISCUSSION THREAD

**Discussion by thinkasany** [ant-design/ant-design#51885] [RFCs]
# Related issues
[feat: ConfigProvider support tooltip](https://github.com/ant-design/ant-design/pull/51872)
[[RFC] Semantic DOM Structure for all Components]( https://github.com/ant-design/ant-design/discussions/40221) 
[ConfigProvider support component className and style](https://github.com/ant-design/ant-design/issues/43051)

# Basic structure order

* section
  * header
    * title
    * extra
  * body
    * content
  * footer

# Semantic DOM

✅可以使用
🔒特殊组件使用
⚠️不推荐使用，理论上不再新增
🚫废弃，勿用
🎨 新加入，待讨论的

| | Name | Description | Components |
| - | ---------- | --------- |--------- |
| ✅ | actions | A group of action button in the component | Card, List |
| ✅ | body | Card-like component body part which includes the padding. | Card, Collapse, Drawer, Modal, Tooltip, Popup, Popconfirm |
| ⚠️ | content | The render wrap of full card. Not the good name in current case | Drawer, Modal |
| ✅ | content | The content under body without padding. Some of component has this layer. | Descriptions |
| ⚠️ | count | Count number node wrapper | Input |
| ✅ | cover | A mask or large image display part | Card |
| ✅ | extra | The component on the logical position end. If Card-like component, always in the `header` | Card, List |
| ✅ | footer | Card-like component footer part which includes the padding. | Drawer, Modal |
| ✅ | handle | Grab-able sub component | Slider |
| ✅ | header | Card-like component header part which includes the padding. | Card, Collapse, Drawer, Modal |
| ✅ | icon | When component has only one Icon. Set on the Icon node **wrapper**. | Button |
| ✅ | indicator | **Tiny** element with fixed position but not provide fully content display. | Badge |
| ⚠️ | input | Input component only. The inner native input element. | Input |
| ⚠️ | item | Sub item component | Space |
| ✅ | label | When using key value pair. The label part. | Descriptions |
| ✅ | mask | The cover mask in full screen | Drawer, Modal |
| ✅ | prefix | Input like component prefix node | Input |
| ⚠️ | rail | Slider component only. The rail of the full size | Slider |
| ✅ | root | The wrap component which not have the real dom but must to use some side effect to handle it. | All |
| ✅ | suffix | Input like component suffix node | Input |
| ⚠️ | textarea | Input component only. The textarea native input element. | Input.TextArea |
| ⚠️ | track | Slider component only. The track element between handle and handle | Slider |
| ⚠️ | tracks | Slider component only. The tracks element between first handle and last handle | Slider |
| ✅ | title | The node not include with padding. If Card-like component, always in the `header` | Card |
| ⚠️ | wrapper | Bad naming. Not to use | Modal |
| 🎨 |   | Let's discuss : ) |  |

# discussion
希望能在这篇 rfc 中敲定最终的 api 命名。

https://github.com/ant-design/ant-design/pull/51872#discussion_r1867496094

![image](https://github.com/user-attachments/assets/ae2dee0a-f264-43f8-96c7-9821005805ea)

**Comment by arvinxx** [comment]
root 和 inner 分别指代啥？语义不是特别清晰。root 还好， inner 这个词，第一眼是看不懂的。

**Reply by thinkasany** [reply]
content 已经被使用了 ```position: relative```,  这个 inner 的话是根据以前 overlayInnerStyle 简化改造而来，需要这个属性去更改 radius 这些常见的修改。https://github.com/react-component/tooltip/blob/master/src/Popup.tsx#L21

当然希望大家能给出更好的语义化名词建议。

![image](https://github.com/user-attachments/assets/bb52c597-659d-4e28-8a76-784ac0f87ef4)

![image](https://github.com/user-attachments/assets/48f97a5d-8027-488e-98bc-8fa426b2d9ef)

**Reply by zombieJ** [reply]
个人看 content 其实不应给 Drawer Modal 的外容器，这个语义有歧义。感觉他们应该起一个新名字。然后 content 还给 body 内部元素
